contracts/pm/mixins/MixinWrappers.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity ^0.4.25;
+// solium-disable-next-line
+pragma experimental ABIEncoderV2;
+
+import "./interfaces/MTicketBrokerCore.sol";
+
+
+contract MixinWrappers is MTicketBrokerCore {
+    /**
+     * @dev Redeems multiple winning tickets. The function will redeem all of the provided
+     * tickets and handle any failures gracefully without reverting the entire function
+     * @param _tickets Array of winning tickets to be redeemed in order to claim payment
+     * @param _sigs Array of sender signatures over the hash of tickets (`_sigs[i]` corresponds to `_tickets[i]`)
+     * @param _recipientRands Array of preimages for the recipientRandHash included in each ticket (`_recipientRands[i]` corresponds to `_tickets[i]`)
+     */
+    function batchRedeemWinningTickets(
+        Ticket[] memory _tickets,
+        bytes[] _sigs,
+        uint256[] _recipientRands
+    )
+        public
+    {
+        for (uint256 i = 0; i < _tickets.length; i++) {
+            redeemWinningTicketNoRevert(
+                _tickets[i],
+                _sigs[i],
+                _recipientRands[i]
+            );
+        }
+    }
+
+    /**
+     * @dev Redeems a winning ticket that has been signed by a sender and reveals the
+     * recipient recipientRand that corresponds to the recipientRandHash included in the ticket
+     * This function wraps `redeemWinningTicket()` and returns false if the underlying call reverts
+     * @param _ticket Winning ticket to be redeemed in order to claim payment
+     * @param _sig Sender's signature over the hash of `_ticket`
+     * @param _recipientRand The preimage for the recipientRandHash included in `_ticket`
+     * @return Boolean indicating whether the underlying `redeemWinningTicket()` call succeeded
+     */
+    function redeemWinningTicketNoRevert(
+        Ticket memory _ticket,
+        bytes _sig,
+        uint256 _recipientRand
+    )
+        internal
+        returns (bool success)
+    {
+        // ABI encode calldata for `redeemWinningTicket()`
+        // A tuple type is used to represent the Ticket struct in the function signature
+        bytes memory redeemWinningTicketCalldata = abi.encodeWithSignature(
+            "redeemWinningTicket((address,address,uint256,uint256,uint256,bytes32,bytes),bytes,uint256)",
+            _ticket,
+            _sig,
+            _recipientRand
+        );
+
+        // Call `redeemWinningTicket()`
+        assembly {
+            // call will return false upon hitting a revert
+            success := call(
```

Comment:
A little out of my depth here - we are doing this assembly call because we don't want revert?

Reply:
Yeah. When you use either a low level call (i.e. invoking a call on an address and generating the calldata yourself) or an assembly based call, if the called function reverts, the call itself will return `false` instead of reverting (see: https://solidity.readthedocs.io/en/v0.4.25/units-and-global-variables.html#address-related). We can use this non-reverting helper function to define a wrapper batch redemption function without changing the actual ticket redemption code.

We could actually use a low level call instead of dropping down to assembly here, but if we end up changing `redeemWinningTicket()` to actually return a value, then we would need to drop down to assembly in order to fetch the return data because the low level call does not give access to the return data. So, I just left the assembly based call which I think is pretty straightforward and self-contained.

Reply:
I see.  That makes sense.  What do you think about keeping track of which ticket failed or emitting an event for failed tickets?  I think in the current workflow, we'll only see successful ticket redemptions through the `WinningTicketRedeemed` event, but failures will be silent.

Reply:
I'm not sure that emitting an additional event is worth the gas cost because given a set of tickets that are passed in for batch redemption a client can check the transaction receipt for `WinningTicketRedeemed` events and then any tickets that do not have an associated event can be marked by the client as having failed